sklearn/tree/tests/test_tree.py

```diff
@@ -1428,19 +1429,12 @@ def check_sparse_parameters(tree, dataset):
     assert_tree_equal(
         d.tree_,
         s.tree_,
-        "{0} with dense and sparse format gave different trees".format(tree),
+        "{0} with dense and sparse format gave different trees".format(tree_type),
     )
     assert_array_almost_equal(s.predict(X), d.predict(X))
 
-
-def check_sparse_criterion(tree, dataset):
-    TreeEstimator = ALL_TREES[tree]
-    X = DATASETS[dataset]["X"]
-    X_sparse = DATASETS[dataset]["X_sparse"]
-    y = DATASETS[dataset]["y"]
-
     # Check various criterion
```

Comment:
Let's keep a separate test but we can parametrize it instead to have an internal `for` loop

Reply:
I've updated but not sure if I'm doing the desired way. Please let me know if it needs further changes.